### PR TITLE
Adds ensembling HOWTO

### LINKED ISSUE #785: Port ensembling HOWTO from old diff based system
And instead, use a standalone doc with tests like in #771

Here is the old (pre-Linen) HOWTO diff, for reference:
https://github.com/google/flax/blob/master/howtos/diffs/ensembling.diff

### PR DESCRIPTION
**Pull request by marcvanzee**
Demo: https://flax.readthedocs.io/en/howto-ensembling/howtos/ensembling.html

Closes #785

### CODE REVIEW

File: docs/howtos/ensembling.rst [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+  for epoch in range(10):
+    rng, input_rng = jax.random.split(rng)
+    optimizer, _ = train_epoch(optimizer, train_ds, input_rng)
+    loss, accuracy = eval_model(optimizer.target, test_ds)
+
+    logging.info('eval epoch: %d, loss: %.4f, accuracy: %.2f', #!
+                epoch, loss, accuracy * 100)
+
+     // handle metrics
```
Comment (avital) [inline docs/howtos/ensembling.rst:272]: Probably remove this line?

File: docs/howtos/ensembling.rst [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+  for epoch in range(10):
+    rng, input_rng = jax.random.split(rng)
+    optimizer, _ = train_epoch(optimizer, train_ds, input_rng)
+    loss, accuracy = eval_model(optimizer.target, test_ds)
+
+    logging.info('eval epoch: %d, loss: %s, accuracy: %s', #!
+                epoch, loss, accuracy * 100)
+
+    // handle metrics
```
Comment (avital) [inline docs/howtos/ensembling.rst:289]: ditto

File: docs/howtos/ensembling.rst [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+Next we simply do the same for the functions ``create_optimizer``, 
+``train_step``, and ``eval_step``. We also make a minor change to 
+``eval_model``, which ensures the metrics are used correctly in the parallel
+setting.
+
+.. codediff::
+  :title_left: Single-model
+  :title_right: Ensemble
+
+  # #!
```
Comment (avital) [inline docs/howtos/ensembling.rst:113]: Is the first `#` necessary? Why not just `#!`?

**Review by avital — APPROVED** [review]
I think this is really good, you explained all of the details nicely in text, and didn't include the initial code blocks by linking to the MNIST example, it really reads well!

I added a couple of really minor points

File: docs/howtos/ensembling.rst (hunk 1 above)
Comment (marcvanzee) [reply docs/howtos/ensembling.rst:272]: Done

File: docs/howtos/ensembling.rst (hunk 3 above)
Comment (marcvanzee) [reply docs/howtos/ensembling.rst:113]: It's a limitation of the current directive, #! will be rewritten to an empty line, and `codeblock` will ignore that. So we need some content on the first line to make sure it aligns nicely with the RHS.

Comment (avital) [reply docs/howtos/ensembling.rst:113]: Ah I see fine for now. It'll become part of the folklore for HOWTO reviewers ;)

**Comment by joaogui1** [comment]
Hey @marcvanzee I think your last commit kind of broke the alignment.
Before:
image

After: 
image

**Comment by avital** [comment]
Thanks for noticing @joaogui1 -- fixed with c9a9e0b. (It's rebuilding ReadTheDocs as we speak, soon to appear on flax.readthedocs.org)

Please feel free to open a new issue or file a PR for any remaining issues. Thanks again for submitting this HOWTO ... 1 year after you originally wanted to? ;)

**Pull request merged**